app/models/tag.rb

```diff
@@ -13,7 +13,7 @@ class Tag < ActiveRecord::Base
   has_many :tag_groups, through: :tag_group_memberships
 
   def self.tags_by_count_query(opts={})
-    q = TopicTag.joins(:tag, :topic).group("topic_tags.tag_id, tags.name").order('count_all DESC')
+    q = Topic.unscoped { TopicTag.joins(:tag, :topic).group("topic_tags.tag_id, tags.name").order('count_all DESC') }
```

Comment:
> When a topic is deleted, maybe it should lose its tags. Thoughts?

I don't think so. That can be a bad surprise for someone. And can't be (easily) undone.

But I understand that this is complicated to implement.

I think we should move to a topic on Discourse to discuss this :smile: